Edict from the bird: "I've run out of jokes"

Incoming5643 opened the pull request:
Poly will now speak a line or emote when pet.

Poly now responds to getting fed crackers by becoming permanently more annoying for the round.

Poly is now much more likely to drop a line from active recall after saying a line. This doesn't actually impact how often poly says something, just how stale his "short term memory" is. Keep in mind that due to the wonders of saycode the line poly drops isn't the same necessarily as the one he said, so the "poly might say this line twice" factor is still higher than the 20% this pull implies. Also this mechanic means it's entirely possible to have roundstart lines be dropped having never been said.

speak_chance is now a proper percentage chance, before it was actually out of 200 for some strange reason. Animals with high speak chances have had their chance halved to preserve their level of chatter (except for poly).

Poly will now keep track of if he died last round and how many rounds in a row he's lived. While he's building on his record he'll have a few special lines he might say at some point near the beginning of the round and also talks FAR more often (to be annoying).

Adds a dumb monty python joke to dead parrots because we literally don't have any.

:cl: Incoming5643
rscadd: Poly will now speak a line or emote when pet.
rscadd: Poly now responds to getting fed crackers by becoming more annoying for the round.
experiment: Every day he is growing stronger
/:cl:

Incoming5643 committed:
Poly will now speak a line or emote when pet.

Poly is now much more likely to drop a line from active recall after saying a line. This doesn't actually impact how often poly says something, just how stale his "short term memory" is. Keep in mind that due to the wonders of saycode the line poly drops isn't the same necessarily as the one he said, so the "poly might say this line twice" factor is still higher than the 20% this pull implies. Also this mechanic means it's entirely possible to have roundstart lines be dropped having never been said.

speak_chance is now a proper percentage chance, before it was actually out of 200 for some strange reason. Animals with high speak chances have had their chance halved to preserve their level of chatter (except for poly).

Poly will now keep track of if he died last round and how many rounds in a row he's lived. While he's building on his record he'll have a few special lines he might say at some point near the beginning of the round and also talks FAR more often (to be annoying).

Adds a dumb monty python joke to dead parrots because we literally don't have any

PKPenguin321 commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -858,11 +864,20 @@
+		speak += pick("...[longest_survival].", "The things I've seen!", "I have lived many lives!", "What are you before me?")
+		speak_chance = 20 //His hubris has made him more annoying/easier to justify killing
+	else if(rounds_survived > 0)
+		speak += pick("...again?", "No, It was over!", "Let me out!", "It never ends!")
```
spooky

PKPenguin321 replied:
also you should add another line: "WAKE ME UP"

Incoming5643 replied:
had to resist that urge

optimumtact replied:
I'll get it

Iamgoofball commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -858,11 +864,20 @@
 	available_channels = list(":e")
 	Read_Memory()
+	if(rounds_survived == longest_survival)
+		speak += pick("...[longest_survival].", "The things I've seen!", "I have lived many lives!", "What are you before me?")
```
make sure these aren't transferred to the speak list

Incoming5643 replied:
They aren't, there's two "pools" of phrases for poly, the speak list and the speech_buffer list.

The speech_buffer is 500 phrases long and is composed entirely of things poly has overheard. He cannot learn from himself.

The speak list is a list of 8 or 9 phrases that poly actively has on hand to quote from.

When poly says something, he picks a phrase from the speak list. Then, 80% of the time, he'll remove a random phrase from the speak list and replace it with another random phrase from the speech_buffer list. As the round progresses any phrases that are in the speak list but not in the speech_buffer list are extincted, and don't survive to the next round. Even if one of these initial phrases DID survive to round end/poly death, poly only remembers speech_buffer, so they're forever locked to conditionals like this (unless some chucklefuck teaches them to poly manually).

KazeEspada replied:
Or two polys next to each other.

Incoming5643 replied:
Barring summon events or badmins that should never happen.

RemieRichards commented on code/modules/mob/living/simple_animal/simple_animal.dm:
```diff
@@ -148,9 +148,9 @@
+/mob/living/simple_animal/proc/handle_automated_speech(var/override)
 	if(speak_chance)
-		if(rand(0,200) < speak_chance)
+		if(rand(0,100) < speak_chance || override)
```
can you just make this a real ```if(prob(speak_chance))``` now then?

neersighted commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -876,11 +891,21 @@
 
 /mob/living/simple_animal/parrot/Poly/proc/Read_Memory()
 	var/savefile/S = new /savefile("data/npc_saves/Poly.sav")
```
Mind making the savefile a define or const?

Incoming5643 replied:
I tried but there's some weird hang ups with trying to define/const a savefile and/or I'm incompetent.

Considering these are both once per round events I don't think it's a big deal.

neersighted commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -876,11 +891,21 @@
+	if(health > 0)
+		S["rounds survived"]	<< rounds_survived + 1
+		if(rounds_survived == longest_survival)
+			S["longest survival"]	<< longest_survival + 1
```
indentation error?

Incoming5643 replied:
nope, if health is 0 there's no way poly has increased his longest survival streak (cause he's dead)

Dorsisdwarf commented:
:+1:

TheNightingale commented:
What happens if you kill Poly but then lazarus him later on?

Incoming5643 commented:
he'll still have the same words and will resave his memory at round end / when he dies again

tkdrg commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -876,11 +891,21 @@
-	S["phrases"] 	>> speech_buffer
+	S["phrases"] 			>> speech_buffer
+	S["rounds survived"]	>> rounds_survived
+	S["longest survival"]	>> longest_survival
```
i think spaces in the save indexes look ugly, why not S["rounds_survived"]

tkdrg commented on code/modules/mob/living/simple_animal/parrot.dm:
```diff
@@ -285,6 +290,8 @@
 			parrot_state |= PARROT_FLEE		//Otherwise, fly like a bat out of hell!
 			drop_held_item(0)
+	if(!stat && M.a_intent == "help")
+		handle_automated_speech(1) //assured speak/emote
```
maybe changelog this

Incoming5643 replied:
Done

Incoming5643 committed:
Feeding poly crackers (existing mechanic) will now boost his speak chance permanently

Some arbitrary renaming of the new savefile vars

makes what was essentially a prob call a prob call.

Incoming5643 committed:
Roll back the shuffle chance for the speech buffer

I wasn't aware this code ran every life tick!

Incoming5643 commented:
Concerns address, polish applied, changelog generated.

Ready for merge.
